client/src/classes/undo-redo-commands/ModifyCommand.ts

```diff
@@ -0,0 +1,95 @@
+import { FabricObjectIdError } from "@/errors/FabricObjectIdError";
+import { UndoRedoCommand } from "@/interfaces/undo-redo-context";
+import { getItemById, updateDimensions } from "@/utils/board/canvasUtils";
+import { AddCommand } from "./AddCommand";
+import { toast } from "sonner";
+
+/**
+ * The class is used in the canvas undo/redo functionality.
+ * It should be used to memorize changes of the modified object and undo/redo these changes.
+ * The idea is simple: store two objects: an old one and a new one. Add them or remove them from canvas as needed.
+ */
+export class ModifyCommand implements UndoRedoCommand {
+  private _canvas: fabric.Canvas;
+  public get canvas(): fabric.Canvas {
+    return this._canvas;
+  }
+
+  private _objectId: string;
+  public get objectId(): string {
+    return this._objectId;
+  }
+
+  private _addObjectBefore: AddCommand;
+  public get addObjectBefore(): AddCommand {
+    return this._addObjectBefore;
+  }
+
+  private _addObjectAfter: AddCommand;
+  public get addObjectAfter(): AddCommand {
+    return this._addObjectAfter;
+  }
+
+  private _handleStyleChange: () => void;
+  public get handleStyleChange(): () => void {
+    return this._handleStyleChange;
+  }
+
+  /**
+   *
+   * @param canvas The canvas where objects are added/removed
+   * @param objectBefore JSON representation of the object before modification
+   * @param objectAfter JSON representation of the object after modification
+   * @param handleStyleChange A callback for handling style changes
+   * @throws `FabricObjectIdError` if `objectBefore` or `objectAfter` does not have an `id`
+   * @throws Error if `objectBefore.id` and `objectAfter.id` do not match
+   */
+  constructor(canvas: fabric.Canvas, objectBefore: any, objectAfter: any, handleStyleChange: () => void) {
+    this._canvas = canvas;
+    this._handleStyleChange = handleStyleChange;
+
+    if (!objectBefore.id) {
+      throw new FabricObjectIdError(objectBefore);
+    }
+    if (!objectAfter.id) {
+      throw new FabricObjectIdError(objectAfter);
+    }
+    if (objectBefore.id !== objectAfter.id) {
+      throw new Error("Ids of objectBefore and objectAfter are not the same");
+    }
+
+    this._addObjectBefore = new AddCommand(canvas, objectBefore);
```

Comment:
Is there a strong reason for using the 'add' prefix in addObjectBefore and addObjectAfter, and why should AddCommand be included there specifically?

Reply:
Well, the `ModifyCommand` is constructed in such a way that it uses two instances of `AddComand`.
The `addObjectBefore` is used to add the old version of our object (with `redo`) and to remove this old version from the canvas (with `undo`).
The `addObjectAfter` is used to add the new version of our object (with `redo`) and to remove this new version from the canvas (with `undo`). 
So I guess that the most descriptive names of these variables should be `addObjectVersionBeforeModification` and 
`addObjectVersionAfterModification`, but they are a tad mouthful 😆

Reply:
Okay, then leave it